shell/common/rasterizer.cc

```diff
@@ -265,9 +268,32 @@ RasterStatus Rasterizer::DoDraw(
   // TODO(liyuqian): in Fuchsia, the rasterization doesn't finish when
   // Rasterizer::DoDraw finishes. Future work is needed to adapt the timestamp
   // for Fuchsia to capture SceneUpdateContext::ExecutePaintTasks.
-  timing.Set(FrameTiming::kRasterFinish, fml::TimePoint::Now());
+  const auto raster_finish_time = fml::TimePoint::Now();
+  timing.Set(FrameTiming::kRasterFinish, raster_finish_time);
   delegate_.OnFrameRasterized(timing);
 
+  const fml::TimePoint latest_frame_target_time =
+      delegate_.GetLatestFrameTargetTime();
```

Comment:
Can this be moved inside the `if`?

Also, the following code might help it be more accurate for the case where we just haven't gotten the vsync notification yet:

```
    if (latest_frame_target_time < raster_finish_time) {
        latest_frame_target_time += frame_budget_millis;
    }
```

Reply:
Good call.